Sources/Jetworking/Client.swift

```diff
@@ -0,0 +1,152 @@
+            self?.handleResponse(data: data, urlResponse: urlResponse, error: error, endpoint: endpoint, completion: completion)
+        }
+
+        dataTask.resume()
+    }
+
+    public func patch<BodyType: Codable, ResponseType>(endpoint: Endpoint<ResponseType>, body: BodyType, _ completion: @escaping (Result<ResponseType, Error>) -> Void) {
+        let request: URLRequest = createRequest(forHttpMethod: .PATCH, andPathComponent: endpoint.pathComponent)
+        let bodyData: Data = try! clientConfiguration.encoder.encode(body)
+        let dataTask = session.uploadTask(with: request, from: bodyData) { [weak self] data, urlResponse, error in
+            self?.handleResponse(data: data, urlResponse: urlResponse, error: error, endpoint: endpoint, completion: completion)
+        }
+
+        dataTask.resume()
+    }
+
+    public func delete<ResponseType>(endpoint: Endpoint<ResponseType>, parameter: [String: Any] = [:], _ completion: @escaping (Result<ResponseType, Error>) -> Void) {
+        let request: URLRequest = createRequest(forHttpMethod: .DELETE, andPathComponent: endpoint.pathComponent)
+        let dataTask = session.dataTask(with: request) { [weak self] data, urlResponse, error in
+            self?.handleResponse(data: data, urlResponse: urlResponse, error: error, endpoint: endpoint, completion: completion)
+        }
+
+        dataTask.resume()
+    }
+
+    private func createRequest(forHttpMethod httpMethod: HTTPMethod, andPathComponent pathComponent: String) -> URLRequest {
+        return .init(
+            url: clientConfiguration.baseURL.appendingPathComponent(pathComponent),
+            httpMethod: httpMethod,
+            headerFields: getHeaderFields()
+        )
+    }
+
+    private func getHeaderFields() -> [String: String] {
+        var headerFields: [String: String] = [
+            "Accept": "application/json",
+            "Content-Type": "application/json"
+        ]
+
+        if
+            let authorizationHeader: [String: String] = getAuthorizationHeader(),
+            let authorizationKey = authorizationHeader.keys.first,
+            let authorizationValue = authorizationHeader.values.first
+        {
+            headerFields[authorizationKey] = authorizationValue
+        }
+
+        return headerFields
+    }
+
+    private func getAuthorizationHeader() -> [String: String]? {
+        let authorizationHeaderKey: String = "Authorization"
+
+        switch clientConfiguration.authenticationMethod {
+        case .none:
+            return nil
+
+        case let .basicAuthentication(username, password):
+            var authString: String = ""
+            let credentialsString = "\(username):\(password)"
+            if let credentialsData = credentialsString.data(using: .utf8) {
+                let base64Credentials = credentialsData.base64EncodedString(options: [])
+                authString = "Basic \(base64Credentials)"
```

Comment:
This should be extracted and tested as well.

Reply:
I will add a ticket to add tests, for now I would leave it as is if that's okay for you.

Reply:
Yep that's ok!